business_objects/project.py

```diff
@@ -60,6 +66,48 @@ def get_label_distribution(
         return values[0]
 
 
+def get_confidence_distribution(
+    project_id: str,
+    labeling_task_id: str,
+    data_slice_id: Optional[str] = None,
+    num_samples: Optional[int] = None,
+) -> List[float]:
+    query_filter = (
+        session.query(RecordLabelAssociation.confidence)
+        .join(
+            LabelingTaskLabel,
+            RecordLabelAssociation.labeling_task_label_id == LabelingTaskLabel.id,
```

Comment:
please with complete join criteria (project_id) without it, it should still work (since the uuid is unique) however with the full join criteria the db optimizer should be able to reduce the data checked for the join significantly (thats why we indexed every fkey :) )

Reply:
implemented